### PR TITLE
Added support for multiple --from-env flags

### LINKED ISSUE #1094: kubectl create secret: Support multiple --from-env-file flags
**What would you like to be added**: Same as https://github.com/kubernetes/kubectl/issues/1033, but for `kubectl create secret` as well. 

**Why is this needed**: These two creation commands both have the flag in question, so it should continue to behave the same way, i.e. it needs to be made repeatable on both.

/assign @lauchokyip 
/triage accepted
/sig cli

### PR DESCRIPTION
**Pull request by lauchokyip**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Added support for multiple --from-env flags

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1094

#### Special notes for your reviewer:
/cc @KnVerey 
/cc @eddiezane 

#### Does this PR introduce a user-facing change?

```release-note
Adding support for multiple --from-env-file flags
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: Added `return` because without it if the test fails it will throw a panic for the next line when we want to dereference the err `err.Error()` but the err will be `nil`. Logically, it makes sense to have `return` there because we want to move on to the next test cases if the current fail.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
Comment (KnVerey) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:387]: This error message changed slightly. Was it intentional or ported over from the configmap code?

```suggestion
			return fmt.Errorf("env secret file cannot be a directory")
```

Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:387]: Thanks for the quick review! I ported the code from configmap 😅  , will make changes now

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: Actually I think I didn't explain it well enough, 
For example if I have a test case
```go
		"foo": {
			secretName:  "foo",
			fromLiteral:    []string{"key1=value1"},
			expectErr:   `foo`,
		},
```
when it will first reach this line, which is fine because we are expecting an error and the err is `nil`
```go
if test.expectErr != "" && err == nil
```
But when it reaches the next line, `err.Error()` will cause a panic because `err` is `nil`.
```go
if test.expectErr != "" && test.expectErr != err.Error() {
```

If I do,
```go
if test.expectErr != "" {
  require.EqualError(t, test.expectErr, err.Error())
} 
require.NoError(t, err)
if !apiequality.Semantic.DeepEqual ... {
}

```

woudln't that cause panic too if err is `nil`?

Comment (KnVerey) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: Ah yes, you're right. You'd need an additional `require.Error(t, err)` as the second line.

Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: If that's the case, do you think `require` is better or `return`? Personally, I think `return` would make the code cleaner.

Comment (KnVerey) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: I find the `require` version more succinct/readable, but it's a personal preference. I'll LGTM either way once the error wording fix is pushed.

Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:529]: sure! let's go with `require` then, I will play around with it and make an update. Thank you

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go
Comment (KnVerey) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:525]: If you want you can simplify this one slightly with require as well, and have it return instead of attempt to run the DeepEqual if the assertion fails. My full suggestion is below but this is also fine as-is.

```go
if test.expectErr == "" {
  require.NoError(err)
  if !apiequality.Semantic.DeepEqual(secret, test.expected) {
    t.Errorf("\nexpected:\n%#v\ngot:\n%#v", test.expected, secret)
  }
} else {
  require.Error(t, err)
  require.EqualError(t, err, test.expectErr)
}

```

**Review by KnVerey — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:525]: Let's simplify it then, will make a change now

**Comment by lauchokyip** [comment]
Thank you @KnVerey for your thoughtful reviews!

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
Comment (apelisse) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:373]: You could have kept that function and call it for each individual file? :-)

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
Comment (apelisse) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:391]: What do you do if you have duplicate keys? Is this properly handled by this function? Specifically if they are coming from different files with the same name in different directories?

**Review by apelisse — COMMENTED** [review]
That seems to only be changing secrets, does that already work for ConfigMaps?

**Comment by lauchokyip** [comment]
> That seems to only be changing secrets, does that already work for ConfigMaps?

Yea there was a PR merge for ConfigMap https://github.com/kubernetes/kubernetes/pull/101646

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:391]: @eddiezane actually asked the same questions 😄 and @KnVerey answered it over here (https://github.com/kubernetes/kubernetes/pull/101646#issuecomment-8827883522)

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:373]: That could be another way to do it! but would that need another change from the configmap file to keep things in sync?

Comment (apelisse) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go:373]: I think it's fine if you leave it like this. thanks

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go:24]: nit: Add a line here please. Convention is:

```
built in

3rd party

k8s
```

**Review by eddiezane — APPROVED** [review]
Thanks for this! One very small nitpick.

/approve
/hold

**Pull request merged**
